From 6df7cdee8e009b83f43dc033c069110933394a9a Mon Sep 17 00:00:00 2001
From: Shen Yong <b00984@freescale.com>
Date: Fri, 25 Jul 2008 12:40:56 +0800
Subject: [PATCH] ENGR00085136 fix system can't wake up from SR mode

system can't wake up from SR mode by interrupt
fix it by remove voltage reducing operation in SR mode

Signed-off-by: Kevin Shen <b00984@freescale.com>
---
 arch/arm/mach-mx3/dvfs_v2.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx3/dvfs_v2.c b/arch/arm/mach-mx3/dvfs_v2.c
index cd54a6e..fa40a24 100644
--- a/arch/arm/mach-mx3/dvfs_v2.c
+++ b/arch/arm/mach-mx3/dvfs_v2.c
@@ -337,8 +337,10 @@ void pmic_voltage_init(void)
 	volt.sw1a = SW1A_1_25V;
 	pmic_power_switcher_set_dvs(SW_SW1A, volt);
 
-	volt.sw1a = SW1A_0_975V;
-	pmic_power_switcher_set_stby(SW_SW1A, volt);
+	if (cpu_is_mx32()) {
+		volt.sw1a = SW1A_0_975V;
+		pmic_power_switcher_set_stby(SW_SW1A, volt);
+	}
 
 	volt.sw1b = SW1A_1_25V;
 	pmic_power_switcher_set_dvs(SW_SW1B, volt);
-- 
1.5.4.4

